File: atcoder/abc236/abc236_d.py
import sys

def calculate():
    if len(pairs) == n:
        affinity = 0
        for pair in pairs:
            affinity ^= affinities[pair[0]][pair[1]]
        return affinity
    
    l = -1
    for i in range(1, 2*n):
        if not used[i]:
            l = i
            break
        
    used[l] = True
    affinity = 0
    for i in range(1, 2*n):
        if not used[i]:
            pairs.append([l, i]) 
            used[i] = True
            affinity = max(affinity, calculate())
            pairs.pop() 
            used[i] = False
        
    used[l] = False
    return affinity

n = int(input())
affinities = [[ -1 for i in range(2*n-1)] for j in range(2*n-1)]
pairs = []
used = [False for i in range(20)]

for i in range(0, 2*n-1):
    values = [int(x) for x in input().split(" ")]
    for j in range(0, len(values)):
        affinities[i][j] = values[j]
print(calculate())

File: atcoder/abc236/abc236_d.cpp
//
//  abc236_d.cpp
//  Problems
//
//  Created by NicolasCardozo on 1/02/22.
//  Copyright © 2022 NicolasCardozo. All rights reserved.
//

#include<bits/stdc++.h>

using namespace std;

int n;
int a[20][20];
vector<pair<int, int>> pairs;
bool used[20];

int calc() {
  if(pairs.size() == n){
    int ret = 0;
    for(auto p : pairs) ret ^= (a[p.first][p.second]);
    return ret;
  }
  
  int l = -1;
  for(int i = 1; i <= 2*n; i++){
    if(!used[i]){
      l = i;
      break;
    }
  }
  used[l] = true;
  int ret = 0;
  for(int i = 1; i <= 2*n; i++){
    if(!used[i]){
        pairs.push_back({l, i});
        used[i] = true;
        ret = max(ret, calc());
        pairs.pop_back();
        used[i] = false;
    }
  }
    
  used[l] = false;
  return ret;
}

int main() {
  cin >> n;
  for(int i = 1; i <= 2*n-1; i++){
    for(int j = i+1; j <= 2*n; j++){
      cin >> a[i][j];
    }
  }
  cout << calc() << endl;
  return 0;
}
